### PR TITLE
add initial 101 class page

### LINKED ISSUE #34: Write EM Development 101 Class
Turn the Dev Class doc we have into a self-guided intro into building a REDCap Development environment with a running REDCap, Dev tools, and a tour of REDCap from an Admin's perspective. Construct it so that it is accessible to a REDCap project builder with no development skills, yet allow a developer to skip quickly to the parts germane to them. 

Crudely, follow these steps:

1. Copy https://github.com/ctsit/redcap_external_module_development_guide/blob/master/docs/development_class.md to the rc_em_dev_101.md
1. Title it "Building a REDCap Development Environment"
1. Remove chit chat about UAMS
1. Move content relevant to a workshop to footnotes

## Required Resources

1. Add preamble to `Required Resources` to say that these are recommended tools. They easiest for the neophyte. If you know what you are doing use whatever you like. 
1. Add git-bash for windows users to downloads.
1. Add links to recommended IDEs and editors i.e., vim.
1. Make hotlinks to Firefox and Chrome dev tools docs.
1. Recommend a sequence to simplify setup for newbies: get github account; git-bash; install github desktop; Configure github desktop for your github account; install your editor; configure github desktop to use your editor, and git-bash; install Docker Desktop

## Pre-req Setup Tasks

1. For `Pre-req Setup Tasks`: move slack stuff to Footnotes; Delete the rest of the section

## REDCap-Specific Setup

1. Instruct student to install RDC via zip file.
1. Add images to help students understand what is happening as we instruct them to open a terminal and run commands.
1. Annotate images with Skitch

## Wrap up

1. Delete everything from "Playtime" down
1. Direct student to RC EM Dev 102 to take tour through REDCap with Admin rights

### PR DESCRIPTION
**Pull request by ChemiKyle**
Closes #34 

Note that this document will still need Luke's additions for Windows devs.

I have some notes I'll add in comments on the code itself.

### CODE REVIEW

File: docs/emd101.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+- GitHub account - [https://github.com/](https://github.com/)
+- GitHub Desktop - [https://desktop.github.com/](https://desktop.github.com/)
+- Docker Desktop - [https://www.docker.com/products/docker-desktop](https://www.docker.com/products/docker-desktop)
+- Text editor or PHP IDE - [VSCode](https://code.visualstudio.com/)/[VSCodium](https://vscodium.com/) or [PHPStorm](https://www.jetbrains.com/phpstorm/), may be best for beginners.
+- redcap-docker-compose - A docker environment tailored for running and developing for REDCap. See [https://github.com/123andy/redcap-docker-compose](https://github.com/123andy/redcap-docker-compose)
+- A REDCap `.zip` file - Ask your local REDCap Admin for `redcap9.3.5.zip` downloaded from the REDCap Community. If you _are_ that REDCap Admin and have access to the community, the download link is [https://community.projectredcap.org/page/download.html](https://community.projectredcap.org/page/download.html). Please be generous with access to the resources your employees and customers need to learn and develop for REDCap while reminding them of the need to not redistribute REDCap software outside the entity that has licensed it.
```
Comment (ChemiKyle) [inline docs/emd101.md:32]: I'd like to move away from referencing a specific REDCap version

File: docs/emd101.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+- Paste this block at the end of the file and save it.
+
+```
+TZ=America/Chicago
+DOCKER_PREFIX=redcap
+WEB_PORT=1935
+MYSQL_PORT=2935
+PHPMYADMIN_PORT=3935
+MAILHOG_PORT=4935
+```
```
Comment (ChemiKyle) [inline docs/emd101.md:55]: Integrating the `make_rdc_instance.sh` script would be nice (though I found the `.env` file needs some manual changes on certain versions due to port numbers)

File: docs/emd101.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+- In GitHub Desktop open a terminal cd'd to the `redcap-docker-compose` folder with the keystroke \<ctrl\>```
+- Change directories into the `rdc` folder with the command `cd rdc`
+- In the terminal start the containers with this command `docker-compose up -d`
```
Comment (ChemiKyle) [inline docs/emd101.md:59]: VSCode launches a terminal already "`cd`'d" to the directory in which you've opened it. I consider getting Windows devs used to git bash to be important, but having them use this builtin terminal may make it a bit easier at the cost of keeping them unaware of the landscape outside the MS garden.

File: docs/emd101.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+<details>
+<summary>Footnotes
+</summary>
+
+## Footnotes
+
+These instructions were adapted from a REDCap Development Class presented at the University of Arkansas Medical School in February 2020. The original document is located [here](development_class).
+
+## Getting the development exercises
```
Comment (ChemiKyle) [inline docs/emd101.md:123]: I think this section should leave the footnotes and be moved to the top of the page for the exercises themselves.

File: docs/emd101.md (hunk 1 above)
Comment (pbchase) [reply docs/emd101.md:32]: > I'd like to move away from referencing a specific REDCap version

Sure, that's fine. Just say:

```suggestion
- A REDCap `.zip` file - Ask your local REDCap Admin for a recent REDCap Zip file downloaded from the REDCap Community. The REDCap Zip files have names like `redcap10.3.3.zip` where '10.3.3' is the REDCap version number. If you _are_ that REDCap Admin and have access to the community, the download link is [https://community.projectredcap.org/page/download.html](https://community.projectredcap.org/page/download.html). Please be generous with access to the resources your employees and customers need to learn and develop for REDCap while reminding them of the need to not redistribute REDCap software outside the entity that has licensed it.
```

File: docs/emd101.md (hunk 3 above)
Comment (pbchase) [reply docs/emd101.md:59]: What if you say something like, "You'll need to open a terminal cd'd to the such-and-such directory. There are multiple ways to do that, but VS Code makes it easy by ..."

File: docs/emd101.md (hunk 2 above)
Comment (pbchase) [reply docs/emd101.md:55]: if they have a back shell outside of docker to run it in, that would be fine.

File: docs/emd101.md (hunk 4 above)
Comment (pbchase) [reply docs/emd101.md:123]: yes, that makes a lot sense. please make that change

File: docs/emd101.md (hunk 2 above)
Comment (ChemiKyle) [reply docs/emd101.md:55]: We'll hold on this until we know that a Windows machine can handle the script

**Comment by ChemiKyle** [comment]
@pbchase I believe this is ready for review now

File: docs/emd101.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+<img class="screenshot" alt="terminal nav and docker-compose up -d" src="/assets/img/terminal-dcup.png"/>
+
+Your Docker containers are now downloading required software and initializing. This step will take some time, especially on the first run.
+- Once the containers are built, wait one more minute for the webpage to finish setting up.
+- Access web-based redcap installer at [http://localhost:1935](http://localhost:1935) You should see this:
+
+![REDCap Docker-Compose Installer](/assets/img/installer.png)
+
+- Select 'Use a local copy of the full zip installer...'. Click Browse, locate redcap9.3.5.zip, and upload it.
```
Comment (pbchase) [inline docs/emd101.md:73]:
```suggestion
- Select 'Use a local copy of the full zip installer...'. Click Browse, locate your redcap zip file (e.g. redcap10.3.3.zip), and upload it.
```

File: docs/guide_for_devs.md [hunk 6]
```diff
@@ -33,7 +35,7 @@ Beyond those in the REDCap Repo, [GitHub](https://github.com) is commonly used b
 
 ### Software Setup and Class Notes
```
Comment (pbchase) [inline docs/guide_for_devs.md:36]:
```suggestion
### Software Setup
```

File: docs/emd101.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+- Paste this block at the end of the file and save it.
+
+```
+TZ=America/Chicago
+DOCKER_PREFIX=redcap
+WEB_PORT=1935
+MYSQL_PORT=2935
+PHPMYADMIN_PORT=3935
+MAILHOG_PORT=4935
```
Comment (pbchase) [inline docs/emd101.md:54]:
```suggestion
- Paste this block at the end of the file and save it. Make changes to reflect your timezone and REDCap version number. You can find a list of valid time zones at [List of tz database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)

The port numbers should reflect your REDCap version number in some way so you can keep your different REDCap installations from colliding and tell them apart. Each of these port numbers needs to be unique. The standard used here compresses the REDCap version number into 4 digits and prefixes it with 1 through 4. Any method that makes unique port numbers and works for you is fine. Be aware that the max allowed port number is 65535.

```
TZ=America/Chicago
DOCKER_PREFIX=redcap
WEB_PORT=11033
MYSQL_PORT=21033
PHPMYADMIN_PORT=31033
MAILHOG_PORT=41033
```

**Pull request merged**